ddtrace/internal/remoteconfig/worker.py

```diff
@@ -144,17 +147,19 @@ def update_product_callback(self, product, callback):
 
     def register(self, product, pubsub_instance, skip_enabled=False):
         # type: (str, PubSub, bool) -> None
+
+        if not isinstance(pubsub_instance, PubSub):
+            raise TypeError("%s is not a PubSub instance" % str(pubsub_instance))
```

Comment:
🤔  what is this for?

Reply:
When I updated this function with your suggestion, many Remote Config (RC) tests started to fail with "AttributeError: 'function' object has no attribute 'stop.'"

https://github.com/DataDog/dd-trace-py/pull/7548/files#diff-344d4f5be7adec7a15bb259a6da69b1a8a4bb49a813484f6eaafbaff56601657L171

It was hard to find the root of the problem, and I found that we are registering functions at some points instead of PubSub instances, but the error is raised at:

```python
    def stop_subscribers(self, join=False):
        # type: (bool) -> None
        """
        Disable the remote config service and drop, remote config can be re-enabled
        by calling ``enable`` again.
        """
        log.debug(
            "[%s][P: %s] Remote Config Poller fork. Stopping  Pubsub services",
            os.getpid(),
            self._parent_id,
        )
        for pubsub in self._client.get_pubsubs():
>           pubsub.stop(join=join)
```

Reply:
Ah I see, but we already have type annotations here, so I don't think we need this check at runtime? We don't do these checks in other places

Reply:
```
We don't do these checks in other places
```
That's true, I'll update the PR, thanks for your suggestion!